pysat/_files.py

```diff
@@ -193,9 +193,11 @@ def __init__(self, inst, directory_format=None, update_files=False,
 
         # Get list of potential data directory paths from pysat. Construct
         # possible locations for data. Ensure path always ends with directory
-        # separator.
+        # separator. The `directory_format` may or may not have fully specified
+        # the data path.
         self.data_paths = [os.path.join(pdir, self.sub_dir_path)
                            for pdir in pysat.params['data_dirs']]
+        self.data_paths.append(self.sub_dir_path)
```

Comment:
Reminder to self about this LOC.

Reply:
Doesn't this line effectively add `./` as an entry to `pysat.params['data_dirs']`?

Reply:
In some cases, but not all.  This was needed for me to specify a custom path that actually worked across the board.  Should we add an if statement here that only appends if `sub_dir_path` isn't `./`?  Or do you have other complaints about specifying a custom path that the user doesn't want saved through `directory_fmt`?

Reply:
Ok, I went and checked the docstring and that isn't clearly explained.  That definitely needs to be updated.

Adding a `data_dir` keyword sounds ok to me.  This is something I frequently want, so I think we should have it.